spec/round_spec.rb

```diff
@@ -6,12 +6,96 @@
 
 RSpec.describe Round do
   it "exists" do
-    round = Round.new
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
     expect(round).to be_instance_of(Round)
   end
 
-  it "contains a deck" do
+  it "has a deck" do 
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    expect(round.deck).to eq(deck)
+  end
+
+  it "starts with no guesses" do 
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    expect(round.guesses).to eq([])
+  end
+
+  it "has a current card" do
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    expect(round.current_card).to eq("3","Hearts")
+  end
+
+  it "has a new guess" do 
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    new_guess = round.record_guess({value: "3", suit: "Hearts"})
+    expect(new_guess).to be_instance_of(Guess)
+    expect(new_guess.class).to eq(Guess)    
+    expect(new_guess.correct?).to eq(true)    
+  end
+
+  it "knows the number of correct guesses when they are correct" do 
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    new_guess = round.record_guess({value: "3", suit: "Hearts"})
+    expect(round.guesses).to eq([new_guess])
+    expect(round.number_correct).to eq(1)
+    expect(round.current_card).to eq(card_2)
+  end
+
+  it "can record an incorrect guess" do 
+    card_1 = Card.new("3","Hearts")
+    card_2 = Card.new("4", "Clubs")
+    deck = Deck.new([card_1, card_2])
+    round = Round.new(deck)
+    new_guess = round.record_guess({value: "3", suit: "Hearts"})
+    new_guess2 = round.record_guess({value: "Jack", suit: "Diamonds"})
+    expect(round.guesses.count).to eq(2)
```

Comment:
I think getting a count is good. I might write another `expect` statement as well to test what the interaction is showing. 

In the interaction pattern when it says
```ruby
pry(main)> round.guesses
#=> [#<Guess:0x00007f972a15c160 @card=#<Card:0x00007f972a227f18 @suit="Hearts", @value="3">, @response="3 of Hearts">]
```
What would this `expect` statement look like?

Reply:
What data type is the return value of round.guesses?